src/Cli/dotnet/commands/dotnet-test/Program.cs

```diff
@@ -118,15 +119,23 @@ private static TestCommand FromParseResult(ParseResult result, string[] settings
                 "-nologo"
             };
 
-            msbuildArgs.AddRange(result.OptionValuesToBeForwarded(TestCommandParser.GetCommand()));
+            var parsedArgs =
+                result.OptionValuesToBeForwarded(TestCommandParser.GetCommand()) // all msbuild-recognized tokens
+                    .Concat(args); // all tokens that the test-parser doesn't explicitly track (minus the settings tokens)
```

Comment:
@baronfel I have updated this part because we should not pass the "settings" arguments to msbuild (at least not directly).